lib/task_manager.js

```diff
@@ -0,0 +1,67 @@
+class TaskManager {
+  constructor() {
+    this.map = {};
+    this.queue;
+    this.buffer = [];
+    this.lastPushedToQueue = 0;
+  }
+
+  async initQueue(maxConcurrentRequests) {
```

Comment:
This might be called in the scope of constructor. as a result we'll get ready to use object

Reply:
p-queue can't be initialised in the constructor

Reply:
Why? what's wrong with this?